src/core/src/service_interfaces/StatusHandler.py

```diff
@@ -795,14 +835,402 @@ def __try_add_error(error_list, detail):
+                    max_package_list_capacity = 126kb - status_file_without_package_list_size 
+                    
+                    if assessment_index is not none:
+                        _substatus_msg_copy = __get_substatus_msg_at_index(truncated_status_file, _index)
+                        _packages_copy = _substatus_msg_copy['patches']
+
+                    if installation_index is not none:
+                        low_pri_index = __get_installation_low_pri_index()
+                        _substatus_msg_copy = __get_substatus_msg_at_index(truncated_status_file, _index)
+                        _packages_copy = _substatus_msg_copy['patches']
+                        
+                        packages_capacity_diff = 0
+                        current_byte_size = __calc_package_payload_size_on_disk(__assessment_packages_truncated)
+                        while status_file_size_in_bytes > 126 and current_byte_size > max_package_list_capacity:
+                            __apply_truncation_process()
+                                __cut_list_and_get_list_capacity()
+                                    __apply_truncation()
+                                    
+                            __recompose_truncated_status_file()                                    
+                                __get_current_complete_status_errors()
+                                __recompose_truncated_msg()
+                                    __recompose_substatus_errors()
+                                    __create_assessment_tombstone_list()
+                                        __create_assessment_tombstone()
+                                    __recreate_assessment_summary_json()
+                           
+                            __recompose_truncated_status_file()                                    
+                                __get_current_complete_status_errors()
+                                __recompose_truncated_msg()
+                                    __recompose_substatus_errors()
+                                    __create_installation_tombstone
+                                    __recreate_installation_summary_json()
+                                    
+                            current_package_list_size -= diff
+                            if current_package_list_size <= max_package_list_capacity:
+                                break
+                            __get_new_size_in_bytes_after_truncation() 
+                    
+                write_with_retry_using_temp_file()  
+        """
+        status_file_size_in_bytes = len(complete_status_file_payload.encode("utf-8"))   # calc complete_status_file_payload byte size on disk
+        truncated_status_file = json.loads(complete_status_file_payload)    # reload payload into python object
```

Comment:
do not name this as truncated_status_file. At this moment, it is not truncated

Reply:
rewrote this logic to write_complete_status_file()